File: 프로그래머스/0/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.cs
using System;

public class Solution 
{
    public int solution(string myString, string pat) 
    {
        // 문자열을 바꾸는거보단 char배열로 바꾼다음 요소를 바꾸는것이 쉽기 때문에 일단 char[]로 바꾸어준다. 
        char[] ch = myString.ToCharArray();
        int answer = 0;
        
        // A는 B로 B는 A로 바꾸는 과정
        for(int i=0; i<ch.Length; i++) {
            if(ch[i] == 'A') {
                ch[i] = 'B';
            }
            else if(ch[i] == 'B') {
                ch[i] = 'A';
            }
        }
        // 다시 string형으로 바꾸기
        string str = new string(ch);

        return answer = (str.Contains(pat)) ? 1 : 0;
    }
}

File: 프로그래머스/0/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

int solution(string myString, string pat) 
{
    for(char &ch : myString)
    {
        if(ch == 'A')
        {
            ch = 'B';
        }
        else if(ch == 'B')
        {
            ch = 'A';
        }
    }
    // 단순하게 search는 문자열이 포함되어있는가?? compare는 특정문자열이 등장하는 index를 체크하면서 포함까지체크
    if(search(myString.begin(), myString.end(), pat.begin(), pat.end()) != myString.end())
    {
        return 1;
    }
    else
    {
        return 0;
    }
}